src/file.cpp

```diff
@@ -197,7 +197,7 @@ namespace
 		for (int i = 0; i < num_bufs; ++i)
 		{
 			DWORD num_read;
-			if (ReadFile(fd, bufs[i].iov_base, bufs[i].iov_len, &num_read, &ol[i]) == FALSE
+			if (ReadFile(fd, bufs[i].iov_base, DWORD(bufs[i].iov_len), &num_read, &ol[i]) == FALSE
```

Comment:
iovec_t should remain compatible with *nix iovec. Where are plenty piece of code that relies on fact that iov_len is size_t.
````c++
file::iovec_t b = { random_data, size_t(16000)};
...
file::iovec_t b = { (void*)data, size_t(size) };
````
so windows specific code should internaly deal with cast